game/src/game/components/VehicleComponent.cpp

```diff
@@ -226,63 +350,24 @@ void VehicleComponent::SetVehicleName(const string& vehicle_name)
                         g_vehicle_name_.c_str());
 }
 
-void VehicleComponent::OnDebugGui()
-{
-    ImGui::Text("Gear: %d", vehicle_.mTransmissionCommandState.gear);
-    ImGui::Text("Steer: %f", vehicle_.mCommandState.steer);
-    ImGui::Text("Throttle: %f", vehicle_.mCommandState.throttle);
-    ImGui::Text("Front Brake: %f", vehicle_.mCommandState.brakes[0]);
-    ImGui::Text("Rear Brake: %f", vehicle_.mCommandState.brakes[1]);
-
-    const vec3 linear_velocity =
-        PxToGlm(vehicle_.mBaseState.rigidBodyState.linearVelocity);
-    gui::ViewProperty("Linear Velocity", linear_velocity);
-
-    const float speed = glm::length(linear_velocity);
-    ImGui::Text("Speed: %f", speed);
-
-    auto vehicle_frame = vehicle_.mBaseParams.frame;
-    const float lat_speed =
-        vehicle_.mBaseState.rigidBodyState.getLateralSpeed(vehicle_frame);
-    const float long_speed =
-        vehicle_.mBaseState.rigidBodyState.getLongitudinalSpeed(vehicle_frame);
-
-    ImGui::Text("Lat Speed: %f", lat_speed);
-    ImGui::Text("Long Speed: %f", long_speed);
-
-    const vec3 angular_velocity =
-        PxToGlm(vehicle_.mBaseState.rigidBodyState.angularVelocity);
-    gui::ViewProperty("Angular Velocity", angular_velocity);
-}
-
-DirectDriveVehicle& VehicleComponent::GetVehicle()
-{
-    return vehicle_;
-}
-
-void VehicleComponent::SetPlayerStateData(PlayerStateData& data)
-{
-    player_data_ = &data;
-}
-
 void VehicleComponent::SetGear(VehicleGear gear)
 {
+    // this is an UNjustifiably long struct name lmao
+    using enum PxVehicleDirectDriveTransmissionCommandState::Enum;
+
```

Comment:
OH wtf this is so cool, I didn't know you could do that with enums/classes?????
very justified tho, I'm all for having long/descriptive names but damn pasting that everywhere in a switch SUCKS

Reply:
yeah! they're fairly new i think? i've been using em right before switches and when the names are longer than the entire lord of the rings trilogy lol